R/thisfile.R

```diff
@@ -0,0 +1,90 @@
+#' @title Determines the path of the currently running script
+#' @description \R does not store nor export the path of the currently running
+#'   script.  This is an attempt to circumvent this limitation by applying
+#'   heuristics (such as call stack and argument inspection) that work in many
+#'   cases.
+#' @details This functions currently work only if the script was \code{source}d,
```

Comment:
I feel like this needs a stronger caveat - i.e. typically this function is not actually want. Instead you should re-think your approach.

Reply:
@hadley Would love to hear instances where re-thinking the approach provides a better solution. This has seemingly come up a few times (setting up devops for automated build of R reports, as well as a robust team approach with R), where it's challenging / impossible to robustly guarantee the working directory stays as we wish.

If we can avoid it, that'd be great.. but I'm probably too new with R to rethink well myself. Any examples of rethinking the approach might help shed some light (and be helpful to end-users if added to the docs above). Thanks!

Reply:
I have no idea what you're doing, so I think it would be easier for you to describe what you're attempting to do.  I'd say 90% of the cases I've seen where people are attempting to get the path of the running it script were ill-advised. Instead, it's usually better to set it outside the context of the script.

Reply:
I don't know what you're trying to do, so I think it would be easier for you to describe what you're attempting to do.  In my experience, attempting to get the path of the running script is ill-advised in the vast majority of cases. Instead, it's usually better to set it outside the context of the script.